pkg/util/helper/binding.go

```diff
@@ -153,12 +163,20 @@ func FetchWorkload(dynamicClient dynamic.Interface, restMapper meta.RESTMapper,
 // EnsureWork ensure Work to be created or updated.
 //nolint:gocyclo
 // Note: ignore the cyclomatic complexity issue to get gocyclo on board. Tracked by: https://github.com/karmada-io/karmada/issues/460
-func EnsureWork(c client.Client, workload *unstructured.Unstructured, clusterNames []string, overrideManager overridemanager.OverrideManager,
-	binding metav1.Object, scope apiextensionsv1.ResourceScope) error {
-	referenceRSP, desireReplicaInfos, err := calculateReplicasIfNeeded(c, workload, clusterNames)
-	if err != nil {
-		klog.Errorf("Failed to get ReplicaSchedulingPolicy for %s/%s/%s, err is: %v", workload.GetKind(), workload.GetNamespace(), workload.GetName(), err)
-		return err
+func EnsureWork(c client.Client, workload *unstructured.Unstructured, clusterNames []string, clusters []workv1alpha1.TargetCluster,
```

Comment:
```suggestion
func EnsureWork(c client.Client, workload *unstructured.Unstructured, clusterNames []string, scheduleResult []workv1alpha1.TargetCluster,
```
Give parameter a significant name.

Reply:
ok